include/spdlog/sinks/android_sink.h

```diff
@@ -0,0 +1,101 @@
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#pragma once
+
+#if defined(__ANDROID__)
+
+#include <mutex>
+#include "base_sink.h"
+#include "../details/null_mutex.h"
+
+#include <android/log.h>
+
+namespace spdlog
+{
+namespace sinks
+{
+/*
+* Android sink (logging using __android_log_write)
+*/
+template<class Mutex>
+class base_android_sink : public base_sink < Mutex >
+{
+public:
+    explicit base_android_sink(std::string tag="spdlog"): _tag(tag)
+    {
+    }
+
+    void flush() override
+    {
+    }
+
+protected:
+    void _sink_it(const details::log_msg& msg) override
+    {
+        const android_LogPriority priority = convert_to_android(msg.level);
+        const int expected_size = msg.formatted.size();
+        const int size = __android_log_write(
+            priority, _tag.c_str(), msg.formatted.c_str()
+        );
+        if (size > expected_size)
+        {
+            // Will write a little bit more than original message
+        }
+        else
+        {
+            throw spdlog_ex("Send to Android logcat failed");
+        }
+    }
+
+private:
+    static android_LogPriority convert_to_android(spdlog::level::level_enum level)
+    {
+        switch(level)
+        {
+            case spdlog::level::trace: return ANDROID_LOG_VERBOSE;
+            case spdlog::level::debug: return ANDROID_LOG_DEBUG;
+            case spdlog::level::info: return ANDROID_LOG_INFO;
+            case spdlog::level::notice: return ANDROID_LOG_INFO;
+            case spdlog::level::warn: return ANDROID_LOG_WARN;
+            case spdlog::level::err: return ANDROID_LOG_ERROR;
+            case spdlog::level::critical: return ANDROID_LOG_FATAL;
+            case spdlog::level::alert: return ANDROID_LOG_FATAL;
+            case spdlog::level::emerg: return ANDROID_LOG_FATAL;
+            case spdlog::level::off: throw spdlog_ex("Unexpected off");
```

Comment:
Not sure about what is this level about, so just throw exception. Let me know if it's not correct.